Content.Client/GameObjects/Components/Photography/PhotoCameraComponent.cs

```diff
@@ -0,0 +1,204 @@
+                case SuicideSelfieMessage selfie:
+                    var viewport = _eyeManager.GetWorldViewport();
+                    var center = _eyeManager.WorldToScreen(_playerManager.LocalPlayer.ControlledEntity.Transform.GridPosition);
+                    TryTakePhoto(selfie.Who, new Vector2(center.X, center.Y), true);
+                    break;
+            }
+        }
+
+        public async void TryTakePhoto(EntityUid author, Vector2 photoCenter, bool suicide = false)
+        {
+            if (!CameraOn)
+            {
+                _notifyManager.PopupMessageCursor(_playerManager.LocalPlayer.ControlledEntity, Loc.GetString("Turn the {0} on first!", Owner.Name));
+                return;
+            }
+
+            if(Film <= 0)
+            {
+                _notifyManager.PopupMessageCursor(_playerManager.LocalPlayer.ControlledEntity, Loc.GetString("No film!"));
+                return;
+            }
+
+            //Play sounds
+            SendNetworkMessage(new TakingPhotoMessage());
+
+            //Take a screenshot before the UI, and then crop it to the photo radius
+            var screenshot = await _clyde.ScreenshotAsync(ScreenshotType.BeforeUI);
+            var cropDimensions = EyeManager.PixelsPerMeter * (Radius * 4);
+
+            //We'll try and center, but otherwise we'll shift the box so it doesn't go outside the viewport
+            var cropX = (int)Math.Clamp(Math.Floor(photoCenter.X - cropDimensions / 2), 0, screenshot.Width - cropDimensions);
+            var cropY = (int)Math.Clamp(Math.Floor(photoCenter.Y - cropDimensions / 2), 0, screenshot.Height - cropDimensions);
+
+            Logger.InfoS("photo", $"cropX:{cropX}, cropY:{cropY}, w:{screenshot.Width}, h:{screenshot.Height}");
+
+            var success = false;
+            var filename = $"MY_PHOTO";
+            using (screenshot)
+            {
+                screenshot.Mutate(
+                    i => i.Crop(new Rectangle(cropX, cropY, cropDimensions, cropDimensions))
+                );
+
+                for (var i = 0; i < 5; i++)
+                {
+                    try
+                    {
+                        if (i != 0)
+                        {
+                            filename = $"MY_PHOTO-{i}";
+                        }
+
+                        await using var file =
+                            _resourceManager.UserData.Open(new ResourcePath("/Screenshots") / $"{filename}.png", FileMode.OpenOrCreate);
+
+                        await Task.Run(() =>
+                        {
+                            screenshot.SaveAsPng(file);
+                        });
+
+                        success = true;
+                        Logger.InfoS("photo", "Photo taken as {0}.png", filename);
+                        break;
+                    }
+                    catch (IOException e)
+                    {
+                        Logger.WarningS("photo", "Failed to save photo, retrying?:\n{0}", e);
```

Comment:
I don't know if this will dump the whole stack trace in logs or not. Might be nicer to just use `e.Message` if it does result in the full trace.

Reply:
Never actually had this occur, but I'll force one and see how it looks.